### PR TITLE
Added flag to determine whether or not to display no value for fields…

### PR DESCRIPTION
**Pull request by Blackbaud-CaseyKour**
… existing on item and not match

### CODE REVIEW

File: src/modules/link-records/link-records-item-diff.component.ts [hunk 1]
```diff
@@ -62,34 +63,34 @@ export class SkyLinkRecordsItemDiffComponent implements OnInit {
...
-      this.state.map((s: any) => s.selected.item)
-      .filter((s: any) => this.selectedByDefault !== undefined)
-      .take(1)
-      .subscribe((selected: any) => {
-        matchFields.forEach(matchField => {
-          if (selected[this.key] && selected[this.key].hasOwnProperty(matchField.key)) {
-            return;
-          }
+      this.state.map(s => s.selected.item)
+        .filter(s => this.selectedByDefault !== undefined)
```
Comment (Blackbaud-LuisBello) [inline src/modules/link-records/link-records-item-diff.component.ts]: the linter didnt complain about implicit type here?

File: src/modules/link-records/link-records-item-diff.component.ts [hunk 2]
```diff
@@ -104,12 +105,15 @@ export class SkyLinkRecordsItemDiffComponent implements OnInit {
 
   get fieldValues() {
     return Observable.combineLatest(
-      this.state.map((s: any) => s.fields.item[this.key] || []).distinctUntilChanged(),
-      this.state.map((s: any) => s.selected.item[this.key] || {}).distinctUntilChanged(),
-      (fields: LinkRecordsFieldModel[], selected: {[key: string]: boolean}) => {
+      this.state.map(s => s.fields.item[this.key] || []).distinctUntilChanged(),
```
Comment (Blackbaud-LuisBello) [inline src/modules/link-records/link-records-item-diff.component.ts]: implicit type?

File: src/modules/link-records/link-records.component.ts [hunk 3]
```diff
@@ -60,6 +60,7 @@ export class SkyLinkRecordsComponent implements OnInit, AfterContentInit, OnDest
   @Input() public itemTitleTemplate: TemplateRef<any>;
   @Input() public keyIdSelector: string = 'id';
   @Input() public selectedByDefault: boolean = true;
+  @Input() public showNewFieldValues: boolean;
```
Comment (Blackbaud-LuisBello) [inline src/modules/link-records/link-records.component.ts]: this should be initialized to true (and so remove the boolean type, itll be inferred when you assign)

**Review by Blackbaud-LuisBello — CHANGES_REQUESTED** [review]

File: src/modules/link-records/link-records-item-diff.component.ts [hunk 4]
```diff
@@ -27,12 +27,13 @@ export class SkyLinkRecordsItemDiffComponent implements OnInit {
   @Input() public match: LinkRecordsMatchModel;
   @Input() public fields: Array<any>;
   @Input() public selectedByDefault: boolean;
+  @Input() public showNewFieldValues = true;
```
Comment (Blackbaud-LuisBello) [inline src/modules/link-records/link-records-item-diff.component.ts]: remove true off this, it bubbles down from the top level. only the top level needs to be initialized

**Review by Blackbaud-LuisBello — APPROVED** [review]

**Review by blackbaud-joshgerdes — APPROVED** [review]

**Pull request merged**